lms/static/js/utils/navigation.js

```diff
@@ -0,0 +1,73 @@
+var edx = edx || {},
+
+    Navigation = (function() {
+
+        var navigation = {
+
+            init: function() {
+
+                if ($('#accordion').length) {
+
+                    navigation.openAccordion();
+                }
+            },
+
+            openAccordion: function() {
+                $('#open_close_accordion a').click();
+                $('.course-wrapper').removeClass('closed');
+                $('#accordion').show();
+
+                navigation.checkForCurrent();
+                navigation.listenForClick();
+            },
+
+            checkForCurrent: function() {
+                var active = $('#accordion div div:has(a.active)').index('#accordion div div') ? $('#accordion div div:has(a.active)').index('#accordion div div') : 0;
+
+                $('#accordion .button-chapter:eq(' + active + ')').trigger('click');
+            },
+
+            listenForClick: function() {
+                $('#accordion').on('click', '.button-chapter', function(event) {
+                    navigation.resetAllAccordions();
+                    navigation.openAccordionSection(event.currentTarget);
+                    navigation.setAriaAttrs(event.currentTarget);
+                });
+            },
+
+            resetAllAccordions: function() {
+                $('.chapter-content-container').hide();
+                $('.chapter-content-container .chapter-menu').hide();
+
+                $('#accordion .button-chapter').each(function(event) {
+                    $(this).removeClass('is-open').attr('aria-pressed', 'false');
+                    $(this).next('.chapter-content-container').attr('aria-expanded', 'false');
+                    $(this).children('.group-heading').removeClass('active');
+                    $(this).children('.group-heading').find('.icon').addClass('fa-caret-right').removeClass('fa-caret-down');
+                });
+            },
+
+            openAccordionSection: function(section) {
+                $(section).next('.chapter-content-container').show().focus();
+                $(section).next('.chapter-content-container').find('.chapter-menu').show();
+
+                navigation.setAriaAttrs(section);
+            },
+
+            setAriaAttrs: function(section) {
+                $(section).addClass('is-open').attr('aria-pressed', 'true');
+                $(section).next('.chapter-content-container').attr('aria-expanded', 'true');
+                $(section).children('.group-heading').addClass('active');
+                $(section).children('.group-heading').find('.icon').removeClass('fa-caret-right').addClass('fa-caret-down');
+            }
```

Comment:
This function does a opposite of liens 43-46, it would make sense to have 2 functions with more meaningful names that take an element and open or close the accordion element